.github/workflows/build_and_run_idf_examples.yml

```
@@ -0,0 +1,75 @@
+name: Build and Run ESP-IDF USB Device examples
+
+on:
+  schedule:
+    - cron: '0 0 * * SAT' # Saturday midnight
+  pull_request:
+    types: [opened, reopened, synchronize]
+
+jobs:
+  build:
+    strategy:
+      matrix:
+        # idf_ver: ["release-v5.0", "release-v5.1", "release-v5.2", "release-v5.3", "release-v5.4", "latest"]
+        idf_ver: ["latest"]
+    runs-on: ubuntu-20.04
+    container: espressif/idf:${{ matrix.idf_ver }}
+    steps:
+      - uses: actions/checkout@v4
+        with:
+          submodules: 'true'
+      - name: Build ESP-IDF USB Device examples
+        shell: bash
+        run: |
+          . ${IDF_PATH}/export.sh
+          pip install idf-component-manager==1.5.2 idf-build-apps==2.4.3 --upgrade
+          export EXAMPLES_PATH="${IDF_PATH}/examples/peripherals/usb/device/tusb_hid"
+          export MANIFEST_FILE="${IDF_PATH}/examples/peripherals/.build-test-rules.yml"
+          python .github/ci/override_managed_component.py tinyusb . ${EXAMPLES_PATH}
+          idf-build-apps find --path ${EXAMPLES_PATH}/ --recursive --target esp32s2 --manifest-file ${MANIFEST_FILE} --build-dir build_@t_@w --work-dir @f_@t_@w
+          idf-build-apps build --path ${EXAMPLES_PATH}/ --recursive --target esp32s2 --manifest-file ${MANIFEST_FILE} --build-dir build_@t_@w --work-dir @f_@t_@w
+      - uses: actions/upload-artifact@v4
+        with:
+          name: usb_device_tusb_apps_bin_${{ matrix.idf_ver }}
+          path: |
+              **/*_usb_device_tusb_*/build_esp*/bootloader/bootloader.bin
+              **/*_usb_device_tusb_*/build_esp*/partition_table/partition-table.bin
+              **/*_usb_device_tusb_*/build_esp*/tusb_*.bin
+              **/*_usb_device_tusb_*/build_esp*/build/flasher_args.json
+              **/*_usb_device_tusb_*/build_esp*/config/sdkconfig.json
```

```suggestion
              ${env.IDF_PATH}/*/**/*_usb_device_tusb_*/build_esp*/bootloader/bootloader.bin
              ${env.IDF_PATH}/*/**/*_usb_device_tusb_*/build_esp*/partition_table/partition-table.bin
              ${env.IDF_PATH}/*/**/*_usb_device_tusb_*/build_esp*/tusb_*.bin
              ${env.IDF_PATH}/*/**/*_usb_device_tusb_*/build_esp*/build/flasher_args.json
              ${env.IDF_PATH}/*/**/*_usb_device_tusb_*/build_esp*/config/sdkconfig.json
```

though it's ugly, I guess it's correct, according to https://github.com/actions/upload-artifact?tab=readme-ov-file#examples

> If a wildcard pattern is used, the path hierarchy will be preserved after the first wildcard pattern